### DISCUSSION TITLE
京东云一代能不能删除无用分区，充分利用32M存储空间

### DISCUSSION THREAD

**Discussion by hyfink** [immortalwrt/immortalwrt#1215] [General]
京东云一代本来内存就只有32M，immortal固件能不能将官方保留的mini、oem分区（mtdblock7、8）等分区删掉，充分利用32M存储。否则安装软件空间真的很紧张。
目前一代也没有任何挖矿价值了。

**Comment by 1715173329** [comment]
我们需要与现有 uboot 以及固件保持兼容，即 不能。

**Reply by hyfink** [reply]
你好，我在网上搜索解决方案，结合自己理解，貌似可以通过修改uboot环境变量：setenv mtdparts mtdparts=<闪存编号>:<分区大小>@0(a),30M(b),-(c)类似的命令重新分区。有几个不确定点想咨询下：
1、因为目前刷的breed，是不是也可以使用mtdparts命令
2、假使上面方案可行的话，也只是将多余的mini（mtd7）和oem（mtd8）合并到firmware（mtd3）分区里，而kernel、rootfs是firmware内部分出来的，最终想扩展的rootfs_data分区又是rootfs里面分出来的。现在不知道怎么修改rootfs_data的分区信息，还是说只要firmware分区扩大了，后面rootfs_data分区会自动占满剩余空间。

**Comment by hyfink** [comment]
昨晚又查了些资料，更新下进展：
原来breed也支持向内核传参，通过设置环境变量：linux.cmdline，值为console=ttyS0,115200 root=/dev/mtdblock5 rootfstype=squashfs,jffs2 mtdparts=spi0.0:192k@0(u-boot)ro,64k@192k(config),64k@256k(factory),-(firmware)。理论上可行。

但是貌似openwrt在编译内核的时候没有开启相应权限，查看内核启动日志：
[    0.000000] Kernel command line: console=ttyS0,115200 rootfstype=squashfs,jffs2
仍然是默认参数。并未使用breed里面自定义的值。

所以暂时应该是无解。除非自行修改并编译源码了。
给后来人做个记录。目前版本23.05.1

**Comment by Erich-Chen** [comment]
在刷入 ImmortalWrt 后，我直接把默认就有的那个 10GB 的分区挂载成 /overlay 了。
